### PR TITLE
Fix the `migrate` subcommand to replace build.gradle correctly.

### LINKED ISSUE #715: The latest Embulk(0.8.26) `migrate` command result differ than Embulk 0.8.17
### Summaries.

The latest Embulk(0.8.26/Un-JRuby version) `migrate` command result differ than Embulk 0.8.17(JRuby version)

* 0.8.17: It updates gradle version **AND** change embulk version in build.gradle files
* 0.8.26: It updates gradle version only. 
  * It doesn't change embulk versions in build.gradle files. 

### Prepare test

```
git clone https://github.com/embulk/embulk-input-jdbc
cd embulk-input-jdbc
```

sha1 hash is `ebfff0b249d507fc730c87e08b56e6aa492060ca` v0.8.4

### Expect behavior(0.8.17)

```
/tmp/embulk-0.8.17.jar migrate .
2017-07-08 12:37:46.342 +0900: Embulk v0.8.17
Detected Java plugin for Embulk 0.8.15...
  Modified /gradle/wrapper/gradle-wrapper.properties
  Modified /gradle/wrapper/gradle-wrapper.jar
  Modified /build.gradle
  Modified /embulk-input-db2/build.gradle
  Modified /embulk-input-mysql/build.gradle
  Modified /embulk-input-oracle/build.gradle
  Modified /embulk-input-postgresql/build.gradle
  Modified /embulk-input-redshift/build.gradle
Done. Please check modifieid files.
```

```diff
diff --git a/build.gradle b/build.gradle
index 2ecb13a..308dab9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -75,8 +75,8 @@ subprojects {
     targetCompatibility = 1.7
 
     dependencies {
-        compile  'org.embulk:embulk-core:0.8.15'
-        provided 'org.embulk:embulk-core:0.8.15'
+        compile  'org.embulk:embulk-core:0.8.17'
+        provided 'org.embulk:embulk-core:0.8.17'
         testCompile 'org.embulk:embulk-test:0.8.15'
     }
 
diff --git a/embulk-input-db2/build.gradle b/embulk-input-db2/build.gradle
index f52ae93..cf2b781 100644
--- a/embulk-input-db2/build.gradle
+++ b/embulk-input-db2/build.gradle
@@ -1,6 +1,6 @@
 dependencies {
     compile project(':embulk-input-jdbc')
 
-    testCompile 'org.embulk:embulk-standards:0.8.15'
+    testCompile 'org.embulk:embulk-standards:0.8.17'
     testCompile files('driver/db2jcc4.jar')
 }
diff --git a/embulk-input-mysql/build.gradle b/embulk-input-mysql/build.gradle
index d0bd073..14a2120 100644
--- a/embulk-input-mysql/build.gradle
+++ b/embulk-input-mysql/build.gradle
@@ -3,5 +3,5 @@ dependencies {
 
     compile 'mysql:mysql-connector-java:5.1.34'
 
-    testCompile 'org.embulk:embulk-standards:0.8.15'
+    testCompile 'org.embulk:embulk-standards:0.8.17'
 }
diff --git a/embulk-input-oracle/build.gradle b/embulk-input-oracle/build.gradle
index 2b2c514..d0ec7fc 100644
--- a/embulk-input-oracle/build.gradle
+++ b/embulk-input-oracle/build.gradle
@@ -1,6 +1,6 @@
 dependencies {
     compile project(':embulk-input-jdbc')
 
-    testCompile 'org.embulk:embulk-standards:0.8.15'
+    testCompile 'org.embulk:embulk-standards:0.8.17'
     testCompile files('driver/ojdbc7.jar')
 }
diff --git a/embulk-input-postgresql/build.gradle b/embulk-input-postgresql/build.gradle
index 136fc09..d489bbe 100644
--- a/embulk-input-postgresql/build.gradle
+++ b/embulk-input-postgresql/build.gradle
@@ -3,5 +3,5 @@ dependencies {
 
     compile 'org.postgresql:postgresql:9.4-1205-jdbc41'
 
-    testCompile 'org.embulk:embulk-standards:0.8.15'
+    testCompile 'org.embulk:embulk-standards:0.8.17'
 }
diff --git a/embulk-input-redshift/build.gradle b/embulk-input-redshift/build.gradle
index e5e756d..561f6d9 100644
--- a/embulk-input-redshift/build.gradle
+++ b/embulk-input-redshift/build.gradle
@@ -4,5 +4,5 @@ dependencies {
 
     testCompile project(':embulk-input-jdbc').sourceSets.test.output
 
-    testCompile 'org.embulk:embulk-standards:0.8.15'
+    testCompile 'org.embulk:embulk-standards:0.8.17'
 }
diff --git a/gradle/wrapper/gradle-wrapper.jar b/gradle/wrapper/gradle-wrapper.jar
index 0087cd3..deedc7f 100644
Binary files a/gradle/wrapper/gradle-wrapper.jar and b/gradle/wrapper/gradle-wrapper.jar differ
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 76b3133..722184f 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Wed Feb 04 13:46:12 PST 2015
+#Sun Jan 08 00:35:58 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.2.1-bin.zip
```

### Actual behavior(0.8.26)

```
2017-07-08 12:42:06.186 +0900: Embulk v0.8.26
Detected Java plugin for Embulk 0.8.15...
Done. Please check modified files.
```

```diff
diff --git a/gradle/wrapper/gradle-wrapper.jar b/gradle/wrapper/gradle-wrapper.jar
index 0087cd3..deedc7f 100644
Binary files a/gradle/wrapper/gradle-wrapper.jar and b/gradle/wrapper/gradle-wrapper.jar differ
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index 76b3133..722184f 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Wed Feb 04 13:46:12 PST 2015
+#Sun Jan 08 00:35:58 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.2.1-bin.zip
```

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
This PR fix #715 issue. 

@dmikurube please take a look when you have a time.

### CODE REVIEW

File: embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java [hunk 1]
```diff
@@ -292,19 +292,23 @@ public FileVisitResult visitFile(Path filePath, BasicFileAttributes attributes)
                             int position = 0;
                             String modifiedData = originalData;
                             while (position < modifiedData.length()) {
-                                final Matcher matcher = pattern.matcher(modifiedData.substring(position));
-                                if (!matcher.matches()) {
+                                final String postModifiedData = modifiedData.substring(0,position);
```
Comment (dmikurube) [inline embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java]: nitpick: a whilespace after `,`.

File: embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java [hunk 2]
```diff
@@ -292,19 +292,23 @@ public FileVisitResult visitFile(Path filePath, BasicFileAttributes attributes)
...
+                                final String preModifiedData = modifiedData.substring(position);
```
Comment (dmikurube) [inline embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java]: 1. `postModifiedData` sounds like "the data **after** `modifiedData`", but the code actually looks like "the earlier part of `modifiedData`". Which one did you intend? If it means like "the earlier part", I'd suggest like:
    * `formerModifiedData` / `latterModifiedData`
    * `earlierModifiedData` / `laterModifiedData`
    * `modifiedDataBeforePosition` / `modifiedDataAfterPosition`
2. Anyway even if we use this `post-` and `pre-`, they sound reversed. `post-` is a kind of "after something", and `pre-` is a kind of "before something".

**Comment by hiroyuki-sato** [comment]
@dmikurube Thank you for reviewing. 

Oh! sorry. It was reverse`pre` and `post`.
I changed the variable name `formerModifiedData` and `latterModifiedData`. 

Please take a look again when you can.

File: embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java (hunk 2 above)
Comment (hiroyuki-sato) [inline embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java]: Change `formerModifiedData` and `latterModifiedData`

File: embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java (hunk 1 above)
Comment (hiroyuki-sato) [inline embulk-cli/src/main/java/org/embulk/cli/EmbulkMigrate.java]: Add whitespace.

**Comment by dmikurube** [comment]
Merging.

**Comment by hiroyuki-sato** [comment]
Thank you!.

**Pull request merged**